openrouteservice/src/main/java/org/heigit/ors/api/requests/matrix/MatrixRequestOptions.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * This file is part of Openrouteservice.
+ *
+ * Openrouteservice is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this library;
+ * if not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.heigit.ors.api.requests.matrix;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import org.heigit.ors.api.requests.common.APIEnums;
+import org.heigit.ors.api.requests.routing.RequestProfileParams;
+import org.heigit.ors.api.requests.routing.RouteRequest;
+import org.heigit.ors.isochrones.IsochroneRequest;
+import org.json.simple.JSONObject;
+
+@ApiModel(value = "Route Options", description = "Advanced options for routing", subTypes = {RouteRequest.class, IsochroneRequest.class})
```

Comment:
These specifications need adjusting to the matrix context.

Reply:
Done

Reply:
Thanks! Normally we would probably want to include `subTypes = {MatrixRequest.class}` in `ApiModel` specification, however, I'm not sure whether this is currently desired considering that we don't expose matrix `options` parameter to the API. Maybe @takb has some additional thoughts on this?

Reply:
Might still make sense to include `subTypes = {MatrixRequest.class}` as long as `options` in `MatrixRequest` are set to hidden.

Reply:
With matrix `options` set to hidden I think it wouldn't harm to include `subTypes = {MatrixRequest.class}` in the `ApiModel` as it will be probably necessary anyway as soon as we decide to expose the options in the public API.

Reply:
Any plans on addressing this? My concern is that if we leave it as interactive docs might fail to render properly once we decide on including matrix `options` in ORS public API.

Reply:
Sorry, I overlooked this. Done.